### PR TITLE
Add Arabic language file

### PR DESCRIPTION
**Pull request by alshakero**
### Context
This adds Arabic translation file for the UI.

### How has this been tested?
I modified the `js/demo` app to test this. This change is still in this PR just to help you test, I can revert it once the UI is tested.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #8760 
2. #9150 

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation - PR of this here: https://github.com/handsontable/handsontable/pull/9213.

### CODE REVIEW

File: examples/next/docs/js/demo/src/index.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ import "pikaday/css/pikaday.css";
 import { data } from "./constants";
 import { progressBarRenderer, starRenderer } from "./customRenderers";
 import "./styles.css";
+import { ar } from 'handsontable/i18n';
+
+Handsontable.languages.registerLanguageDictionary(ar);
```
Comment (alshakero) [inline examples/next/docs/js/demo/src/index.js:10]: This change is temporary to help test the PR.

File: handsontable/src/i18n/languages/ar.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * Authors: Handsoncode
+ * Last updated: Feb 9, 2022
+ *
+ * Description: Definition file for Arabic - Without a specific country.
+ */
+import * as C from '../constants';
+
+const dictionary = {
+  languageCode: 'ar',
+  languageDirection: 'rtl',
```
Comment (alshakero) [inline handsontable/src/i18n/languages/ar.js:12]: This doesn't do anything as of now.

**Comment by AMBudnik** [comment]
I'm more than glad to finally see Arabic on the list. Thank you for the contribution.

File: handsontable/src/i18n/languages/ar.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ * @preserve
+ * Authors: Handsoncode
+ * Last updated: Feb 9, 2022
+ *
+ * Description: Definition file for Arabic - Without a specific country.
+ */
+import * as C from '../constants';
+
+const dictionary = {
+  languageCode: 'ar',
```
Comment (budnix) [inline handsontable/src/i18n/languages/ar.js:11]:
```suggestion
  languageCode: 'ar-AR',
```
In reference to the best practices for language tags ([BCP 47](https://www.rfc-editor.org/rfc/rfc5646.html)), may we agree to use the `ar-AR` tag here?

File: handsontable/src/i18n/languages/index.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ import ptBR from './pt-BR';
 import ruRU from './ru-RU';
 import zhCN from './zh-CN';
 import zhTW from './zh-TW';
+import ar from './ar';
```
Comment (budnix) [inline handsontable/src/i18n/languages/index.js:17]:
```suggestion
import arAR from './ar-AR';
```

File: handsontable/src/i18n/languages/index.js [hunk 5]
```diff
@@ -31,5 +32,6 @@ export {
   ptBR,
   ruRU,
   zhCN,
-  zhTW
+  zhTW,
+  ar
```
Comment (budnix) [inline handsontable/src/i18n/languages/index.js:36]:
```suggestion
  arAR
```

File: docs/next/guides/internationalization/language.md [hunk 6]
```diff
@@ -112,6 +112,7 @@ Below you'll find a list of features which can be translated:
 
 By default, Handsontable uses the **English - United States** language-country set (`en-US` code) for creating the text of UI elements. However, it can be used like every extra, "non-standard" language file, thus the `en-US.js` file can be found in `/dist/languages`, `/languages` and `/src/languages` folders. Currently, we also distribute extra language-country files:
 
+* `ar.js` for **Arabic - Global** (`ar` code).
```
Comment (budnix) [inline docs/next/guides/internationalization/language.md:115]:
```suggestion
* `ar-AR.js` for **Arabic - Global** (`ar-AR` code).
```

File: examples/next/docs/js/demo/src/index.js [hunk 7]
```diff
@@ -5,6 +5,9 @@ import "pikaday/css/pikaday.css";
 import { data } from "./constants";
```
Comment (budnix) [inline examples/next/docs/js/demo/src/index.js:5]: Can I ask you to revert that file? AFAIK the "demo" from the examples project is currently reserved for LTR. Probably something will be changed but only after the official RTL support release.

**Review by budnix — CHANGES_REQUESTED** [review]
Thanks for the contribution 👏  Can I ask you to change the base branch to `rtl` (not `rtl-docs`)?

**Comment by alshakero** [comment]
Alright, I addressed all feedback and move documentation to [another PR](https://github.com/handsontable/handsontable/pull/9213) here.

**Comment by AMBudnik** [comment]
Thank you for all your feedback, @alshakero 
v12.0 with your changes is live!

**Pull request merged**